src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertRegularStringToInterpolatedStringRefactoringProvider.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.ConvertToInterpolatedString
+{
+    /// <summary>
+    /// Code refactoring that converts a regular string containing braces to an interpolated string
+    /// </summary>
+    internal abstract class AbstractConvertRegularStringToInterpolatedStringRefactoringProvider<TExpressionSyntax> : CodeRefactoringProvider
+        where TExpressionSyntax : SyntaxNode
+    {
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var literalExpression = await context.TryGetRelevantNodeAsync<TExpressionSyntax>().ConfigureAwait(false);
+            if (literalExpression == null || !IsAppropriateLiteralKind(literalExpression))
+                return;
+
+            var stringToken = literalExpression.GetFirstToken();
+            if (stringToken.Text is null || (!stringToken.Text.Contains("{") && !stringToken.Text.Contains("}")))
+                return;
+
+            var (document, _, cancellationToken) = context;
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            if (syntaxFacts is null)
+                return;
+
+            // If there is a const keyword, do not offer the refactoring (an interpolated string is not const)
+            var declarator = literalExpression.FirstAncestorOrSelf<SyntaxNode>(syntaxFacts.IsVariableDeclarator);
+            if (declarator != null)
+            {
+                var generator = SyntaxGenerator.GetGenerator(document);
+                if (generator.GetModifiers(declarator).IsConst)
```

Comment:
this unfortunatley isn't enough.  There are other constant locations (like attribute arguments).  Until C# has constant interpolated strings, this wouldn't work there.

*That said*.  This is a refactoring, so the worst that will happen is the user will be offered something non-viable.  THat's not the end of the world.   We do have a helper though that can be useful here called IsConstantExpressionContext.  You could use this to make sure this feature isn't offered in those places.

Reply:
You're absolutely right. I temporarily added a test (not committed) for a string attribute argument, and it failed. I wanted to try out `IsConstantExpressionContext` instead, but it does not appear to be accessible... Any further suggestions would be appreciated.

Reply:
you may need to call into the subclasses so they can reference the existing VB and C# versions of these methods.

Reply:
hrmm... VB may not even have a helper for this.  that will make things more challenging :)

Reply:
i'd say that we just check for common scenarios here.  i.e. walk upwards and see if you're on a constnat (like this), or in an attribute.  and then disable for that.

Reply:
Just pushed a new commit. LMK what you think.